windows/CodePush.cs

```diff
@@ -0,0 +1,668 @@
+
+                        if (lifecycleEventListener == null)
+                        {
+                            // Ensure we do not add the listener twice.
+                            lifecycleEventListener = new CodePushResumeListener(this);
+                            reactContext.AddLifecycleEventListener(lifecycleEventListener);
+                        }
+                    }
+
+                    promise.Resolve("");
+                };
+
+                Context.RunOnNativeModulesQueueThread(installUpdateAction);
+            }
+
+            [ReactMethod]
+            public void isFailedUpdate(string packageHash, IPromise promise)
+            {
+                promise.Resolve(codePush.IsFailedHash(packageHash));
+            }
+
+            [ReactMethod]
+            public void isFirstRun(string packageHash, IPromise promise)
+            {
+                Action isFirstRunAction = async () =>
+                {
+                    bool isFirstRun = codePush.didUpdate
+                        && packageHash != null
+                        && packageHash.Length > 0
+                        && packageHash.Equals(await codePush.codePushPackage.GetCurrentPackageHash());
+                    promise.Resolve(isFirstRun);
+                };
+
+                Context.RunOnNativeModulesQueueThread(isFirstRunAction);
+            }
+
+            [ReactMethod]
+            public void notifyApplicationReady(IPromise promise)
+            {
+                codePush.RemovePendingUpdate();
+                promise.Resolve("");
+            }
+
+            [ReactMethod]
+            public void restartApp(bool onlyIfUpdateIsPending)
+            {
+                Action restartAppAction = async () =>
+                {
+                    // If this is an unconditional restart request, or there
+                    // is current pending update, then reload the app.
+                    if (!onlyIfUpdateIsPending || codePush.IsPendingUpdate(null))
+                    {
+                        await LoadBundle();
+                    }
+                };
+
+                Context.RunOnNativeModulesQueueThread(restartAppAction);
+            }
+
+            [ReactMethod]
+            // Replaces the current bundle with the one downloaded from removeBundleUrl.
+            // It is only to be used during tests. No-ops if the test configuration flag is not set.
+            public void downloadAndReplaceCurrentBundle(String remoteBundleUrl)
+            {
+                // TODO implement this
+            }
+        }
+
+        public IReadOnlyList<Type> CreateJavaScriptModulesConfig()
+        {
+            return new List<Type>();
+        }
+
+        public IReadOnlyList<INativeModule> CreateNativeModules(ReactContext reactContext)
+        {
+            List<INativeModule> nativeModules = new List<INativeModule>();
```

Comment:
We could use the [collection initializer](https://msdn.microsoft.com/en-us/library/bb384062.aspx#Anchor_2) syntax to clean this up a little bit.